test/path_matcher_test.cc

```diff
@@ -355,23 +384,31 @@ TEST_F(PathMatcherTest, PercentEscapesNotUnescapedForMultiSegment2) {
 }
 
 TEST_F(PathMatcherTest, OnlyUnreservedCharsAreUnescapedForMultiSegmentMatch) {
-  MethodInfo* a__c = AddGetPath("/a/{x=**}/c");
-  Build();
+  // All %XX are reserved characters, they should be intact.
+  MultiSegmentMatchWithReservedCharactersBase(
+      "%21%23%24%26%27%28%29%2A%2B%2C%2F%3A%3B%3D%3F%40%5B%5D");
+}
 
-  EXPECT_NE(nullptr, a__c);
+TEST_F(PathMatcherTest,
+       OnlyUnreservedCharsAreUnescapedForMultiSegmentMatchAlwaysDecode) {
+  SetFullyDecodeReservedExpansion(true);
+  // All %XX are reserved characters, they should be decoded.
```

Comment:
comment is wrong, slash is not be decoded

Reply:
Thanks, updated